### PR TITLE
upgrade controller-tools version from v0.2.5 to v0.3.0

### PR DESCRIPTION
**Pull request by camilamacedo86**
**Description**
upgrade controller-tools version from v0.2.5 to v0.3.0

**Motivation**
- avoid technical debts issues
- get bug fix and improvements to support CRD and Webhook api V1. 

PS.:  Has no breaking changes. See: https://github.com/kubernetes-sigs/controller-tools/releases

### CODE REVIEW

**Comment by mengqiy** [comment]
After reading the log, it seems the netlify failed to build due to a too old golang was used.
It seems there is a way to use a newer golang version: https://community.netlify.com/t/go-version-1-13/5680
@camilamacedo86 can you try to fix this issue?

File: docs/book/install-and-build.sh [hunk 1]
```diff
@@ -56,9 +67,10 @@ ${cmd} /tmp/mdbook.${ext}
 chmod +x /tmp/mdbook
 
 echo "grabbing the latest released controller-gen"
-go get sigs.k8s.io/controller-tools/cmd/controller-gen@v0.2.5
+go get sigs.k8s.io/controller-tools/cmd/controller-gen@v0.3.0
 
 # make sure we add the go bin directory to our path
+echo "adding the go bin directory to our path"
```
Comment (estroz) [inline docs/book/install-and-build.sh:73]: Unrelated change.

Comment (camilamacedo86) [reply docs/book/install-and-build.sh:73]: required to fix the CI preview after the change and allow the merge
but you are right. I will split into another PR.
